### PR TITLE
Twig 3.x

### PR DESCRIPTION
**Pull request by nVuln**
composer & creator event as twig extension

### CODE REVIEW

File: src/Node/EventNode.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+use Twig\Compiler;
+use Twig\Node\Node;
+use Illuminate\View\View;
+
+class EventNode extends Node
+{
+    public function compile(Compiler $compiler)
+    {
+        $compiler->write("if (!isset(\$context['__env']) && method_exists(\$this->env, 'mergeShared')) {\n")
```
Comment (filcius) [inline src/Node/EventNode.php:18]: Why do you need mergeShared? I removed mergeShared in favor of an extension that added global constant using the regular Twig way.

**Review by filcius — CHANGES_REQUESTED** [review]
Made a quick review. I will review it more thoroughly and test it more in the next days.

Thank you for your input!

File: src/Node/EventNode.php [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            ->indent()
+            ->write("\$context = \$this->env->mergeShared(\$context);\n")
```
Comment (filcius) [inline src/Node/EventNode.php:20]: Could $context may conflict with a variable appended by another Node? I will check compiled template code to make sure...

File: src/Node/EventNode.php (hunk 1 above)
Comment (nVuln) [reply src/Node/EventNode.php:18]: my mistake, I make this PR before I got your new commit

**Comment by nVuln** [comment]
I think body node is not a good idea
I'm look into display_start node

**Pull request closed without merging**